test/com/google/javascript/jscomp/CollapsePropertiesTest.java

```diff
@@ -233,11 +320,36 @@ public void testAliasCreatedForObjectDepth2_2() {
          "var a$b = {c: 0}; for (var p in a$b) { e(a$b[p]); }");
   }
 
+  public void testEnumDepth1() {
+    test("/** @enum */ var a = {b: 0, c: 1};",
+         "var a$b = 0; var a$c = 1;");
+
+    test("/** @enum */ var a = { /** @nocollapse */ b: 0, c: 1};",
```

Comment:
It may be possible to specifically deny `@nocollapse` annotations on enums. I'll look into that - I don't see any use cases that would require `@nocollapse` on an enum.